GSA/digital-gov-drupal
# Digital-195: SSO Configuration

## Jira ticket

[DIGITAL-195](https://cm-jira.usa.gov/browse/DIGITAL-195)

## Purpose

GSA Auth prefers to use OIDC over SAML, so we're changing to a different module.

* Remove the SAML Auth module
* Add the OpenID Connect module
* Configure both preprod & prod clients
* Set up the preprod as the default, prod is the exception.
* Add the noreqnewpass module to disable the password reset form
* Disable account self-registration

## Deployment and testing

### Local Setup

Standard reseed is sufficient.

### QA/Testing instructions

**UNTESTED! PENDING PROVISIONING BY GSA AUTH**

1. Create an account for yourself with `drush user:create jq.dev@gsa.gov --mail jq.dev@gsa.gov` (use your REAL GSA email)
2. Visit the site as an anonymous user
3. Observe no "reset password" or "register" links
4. Click "Login with GSA Auth"
5. Authenticate to GSA Auth
6. You should be logged in.


## Checklist for the Developer




## Checklist for the Peer Reviewers
